### PR TITLE
Initial dispatch refactoring in Swift

### PR DESCRIPTION
**Pull request by bernardnormier**
This is an initial refactoring of the dispatch pipeline in Swift. It includes the following:

- update the Current class, which used to be generated from local Slice
Most of its properties are now immutable, and adapter is now non-nullable.

- update / rename the main dispatcher bridge between Swift and C++. It's now called SwiftDispatcher and implements Ice C++'s Ice::Object directly - no more Blobject.

- update the line numbers in the exception classes to be an Int32 instead of Int

- simplify the dispatch exception propagation

It was a very convoluted: `Error -> ICERuntimeException exception hierarchy -> std::exception_ptr`. It's now a much simpler `Error -> single ICEDispatchException` and ICEDispatchException holds a `std::exception_ptr`. Nevertheless, this simpler code is temporary since the Swift code should actually marshal the full response, and not create a std::exception_ptr that the C++ code marshals.

- removed the interceptor test (interceptor = dispatch interceptor)

### CODE REVIEW

File: swift/src/Ice/Incoming.swift [hunk 1]
```diff
@@ -224,81 +217,40 @@ public final class Incoming {
...
-      e.file = exception.file
-      e.line = Int32(exception.line)
-      e.unknown = "\(exception)"
-      return e
-    case let exception as UserException:
-      let e = ICEUnknownUserException()
-      e.unknown = "\(exception.ice_id())"
-      return e
-    // 3. Unknown exceptions
+    // OperationNotExistException and friends
```
Comment (bernardnormier) [inline swift/src/Ice/Incoming.swift:227]: I am not fond of our current swift-format configuration. We should fix it.

**Review by pepone — APPROVED** [review]

File: swift/src/IceImpl/BlobjectFacade.h [hunk 2]
```diff
@@ -27,25 +30,21 @@ ICEIMPL_API @protocol ICEBlobjectFacade
...
-           exception:(ICEBlobjectException)exception;
+        sendResponse:(ICESendResponse)sendResponse;
 - (void)facadeRemoved;
 @end
 
 #ifdef __cplusplus
 
-class BlobjectFacade : public Ice::BlobjectArrayAsync
+/// A C++ dispatcher that dispatches requests to a Swift object via an ObjC/Swift "facade".
+class SwiftDispatcher final : public Ice::Object
```
Comment (externl) [inline swift/src/IceImpl/BlobjectFacade.h:40]: Should this file be renamed if we're renaming the class.

**Review by externl — APPROVED** [review]
Looks good!

File: swift/src/IceImpl/BlobjectFacade.h (hunk 2 above)
Comment (bernardnormier) [reply swift/src/IceImpl/BlobjectFacade.h:40]: Yes, in a follow-up PR.

**Pull request merged**